src/b64.ml

```diff
@@ -16,77 +19,143 @@
  *
  *)
 
-let default_alphabet =
-  "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"
+let () = Printexc.record_backtrace true
 
-let uri_safe_alphabet =
-  "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_"
+type alphabet =
+  { emap : int array
+  ; dmap : int array }
 
-let padding = '='
+let (//) x y =
+  if y < 1 then raise Division_by_zero ;
+  if x > 0 then 1 + ((x - 1) / y) else 0
+[@@inline]
 
-let of_char ?(alphabet = default_alphabet) x =
-  if x = padding then 0 else String.index alphabet x
+external unsafe_get_uint8 : string -> int -> int = "%string_unsafe_get" [@@noalloc]
+external unsafe_set_uint8 : bytes -> int -> int -> unit= "%bytes_unsafe_set" [@@noalloc]
+external unsafe_set_uint16 : bytes -> int -> int -> unit = "%caml_string_set16u" [@@noalloc]
```

Comment:
This seems equivalent to `String.unsafe_get |> Char.int` (both of which are still builtins due to Char.int being identity).  Is is better to just reuse those rather than access the externals directly?